### PR TITLE
GoogleCloudMonitoring: Adapt frontend to the new API format

### PR DESCRIPTION
**Pull request by andresmgot**
**What is this feature?**

This first PR adds the new types and the migration code to populate the query in the new format. For the moment the new format is not used anywhere, I will be migrating things little by little (if possible).

Unfortunately, the migration code needs to be duplicated in the frontend and in the backend :disappointed: 

**Which issue(s) does this PR fix?**:

Ref #60158

**Special notes for your reviewer**:

### CODE REVIEW

**Review by aangelisc — APPROVED** [review]
This is great work on the migrations, and greatly simplified. My one request would be to add the following 3 tests:

- A test for the migration of filters in the time series list query
- A test for the migration of preprocessor in the time series list query
- A test for the migration of a time series query (MQL)

**Comment by andresmgot** [comment]
> My one request would be to add the following 3 tests

Makes sense, I thought that this is extensively tested in the backend but it's not justification enough to not test this here.

BTW, I am not going to merge this PR and use this branch to merge other dependendant PRs, that way the temporary code does not reach `main` (even if it's harmless atm).

**Comment by aangelisc** [comment]
> BTW, I am not going to merge this PR and use this branch to merge other dependendant PRs, that way the temporary code does not reach `main` (even if it's harmless atm).

Good idea, completely fine with this!

File: public/app/plugins/datasource/cloud-monitoring/datasource.test.ts [hunk 1]
```diff
@@ -25,4 +26,144 @@ describe('Cloud Monitoring Datasource', () => {
...
+              editorMode: EditorMode.MQL,
+            },
+          },
+          expected: {
+            projectName: 'project',
+            query: 'test query',
+          },
+        },
+      ].forEach((t) =>
+        it(t.description, () => {
```
Comment (aangelisc) [inline public/app/plugins/datasource/cloud-monitoring/datasource.test.ts:155]: 💯 perfect!

**Review by aangelisc — APPROVED** [review]

**Comment by andresmgot** [comment]
:crossed_fingers:

**Pull request merged**